pkg/rke2/workload_cluster.go

```diff
@@ -266,6 +270,16 @@ func (w *Workload) ClusterStatus() ClusterStatus {
 		}
 	}
 
+	// find the rke2-serving secret
+	key := ctrlclient.ObjectKey{
+		Name:      rke2ServingSecretKey,
+		Namespace: metav1.NamespaceSystem,
+	}
+	err := w.Client.Get(ctx, key, &corev1.Secret{})
+
+	// In case of error we do assume the control plane is not initialized yet.
+	status.HasRKE2ServingSecret = err == nil
```

Comment:
```suggestion
        if err != nil {
	    logger := log.FromContext(ctx)
	    logger.Error(err, "Could not get rke2 serving secret")
	}
	status.HasRKE2ServingSecret = err == nil
```

Reply:
This may be a bit too spammy. 
Also this is not an error, in most cases we are just waiting here for the control plane to initialize.

If we want to emit a log, then I'd say it should be Info level, not even warning, because this will happen all times.

Reply:
```suggestion
        if ctrlclient.IgnoreNotFound(err) != nil {
	    logger := log.FromContext(ctx)
	    logger.Error(err, "Unable to get RKE2 serving secret")
	}
	status.HasRKE2ServingSecret = err == nil
```
Probably for not found errors it might be spamming, can also be extended with other frequent err situations, but any other kind after the management cluster client was established should be treated as error. Leaving on your consideration, just think that converting error to bool without context can be difficult to track, and occasional *my CP is unavailable* issues will be more frequent due to failure of child cluster API server.

Reply:
I think the most common case that we will encounter in here is connection issues. This is the case where we just defined a cluster. The control-plane Machine may take a minute maybe to be created, then we have to wait for the physical machine to be bootstrapped, all processes running, kubelet alive and available. 

The entire process will take minutes maybe, during all this time this will fail due to connection errors, however this is to be expected. 

I do see your point however and I agree this should be highlighted to the end user. 
In that case I would just emit a log at Info level. We don't know if this is an error, it's for the user to judge. They will experience that the provider can't connect to the control plane (and therefore continue with the machines rollout). Maybe the user failed to configure the network properly, maybe we just have to wait, we don't know.

Something like this should work:

```go
        if err != nil {
	    logger := log.FromContext(ctx)
	    logger.Info("Control Plane is not initialized yet: %s", err.Error())
	}
```